daal4py/sklearn/_utils.py

```diff
@@ -68,21 +71,18 @@ def set_idp_sklearn_verbose():
         )
 
 
-def daal_check_version(rule):
+def get_daal_version() -> DaalVersionTuple:
+    return int(dv()[0:4]), str(dv()[10:11]), int(dv()[4:8])
+
+
+def daal_check_version(required_version: Tuple[Any, ...]) -> bool:
     # First item is major version - 2021,
-    # second is minor+patch - 0110,
-    # third item is status - B
-    target = (int(dv()[0:4]), dv()[10:11], int(dv()[4:8]))
-    if not isinstance(rule[0], type(target)):
-        if rule > target:
-            return False
-    else:
-        for rule_item in rule:
-            if rule_item > target:
-                return False
-            if rule_item[0] == target[0]:
-                break
-    return True
+    # Second item is status - B
+    # Third is minor+patch - 0110,
+    if isinstance(required_version[0], (list, tuple)):
+        return any(map(daal_check_version, required_version))
```

Comment:
its more concise but I have no idea what's going on here ;)

Reply:
I have provided a better description and a `test_utils.py` that shows how it's supposed to work. Hope that helps. Just need to verify the `test_utils.py` actually runs in CI.